packages/kbn-interpreter/src/public/socket.js

```diff
@@ -18,14 +18,11 @@
  */
 
 import io from 'socket.io-client';
-import { functionsRegistry } from '../common/lib/functions_registry';
-import { getBrowserRegistries } from './browser_registries';
 
 const SOCKET_CONNECTION_TIMEOUT = 5000; // timeout in ms
-let socket;
 
-export async function createSocket(basePath) {
-  if (socket != null) return socket;
+export async function createSocket(basePath, functionsRegistry) {
+  let socket;
```

Comment:
I don't think this makes a lot of sense anymore. Previously, the socket was created once and reused, so it needed to be set to null so that future calls could create a new socket.

Now, every time you call `createSocket`, you get a new socket client, so there's no need to re-assign it.

Also, a question; is it safe to allow more than a single socket client? It wasn't possible before, and now it is...

Reply:
hmm, interesting question ....
- what would happen before with two users using kibana ? one socket per user right ? 
-- so two sockets from single user should behave the same ... ?

so it might work ... never tested however. for example:
- what happens if with each socket we would have a different list of functions in browser ?

Reply:
> what would happen before with two users using kibana ? one socket per user right ?

That's correct, every user gets their own socket client instance.

> what happens if with each socket we would have a different list of functions in browser ?

Good question, it's not a situation we've had to deal with before, but we may have to soon. I was thinking that every app would have its own client, so there would only be a single client instance. Maybe you had another model in mind?

Reply:
didn't had any specific in mind was just wondering if something was planned already. thanks for clarification,.